build.yaml

```diff
@@ -168,14 +196,28 @@ common_cli:
     # https://gitlab.com/gitlab-org/gitlab-ci-multi-runner/issues/20
     # So we will place the virtualenv in /tmp/v instead
     #
-    sudo rm -rf \$venv
-    sudo rm -f results.xml
-    sudo rm -f trial.log
-    sudo rm -rf _trial_temp/
+    if [ "\$(is_osx)" == "true" ]; then
+      # OSX doesn't allow us to sudo (no tty)
+      rm -f results.xml
+      rm -f trial.log
+      rm -rf _trial_temp/
+      rm -rf \${venv}
```

Comment:
:exclamation: It would be nice to get rid of this duplication here (have a list of files to delete?).

I'm also curious how this relates to https://github.com/ClusterHQ/flocker/blob/flocker-create-jenkins-osx-job-FLOC-2854/jobs.groovy.j2#L124-L137 which seems to serve the same purpose.